### PR TITLE
make mypy linting be more consistent with core and fix lints

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #587
* __->__ #591

### CODE REVIEW

**Review by kwen2501 — APPROVED** [review]
Thanks a thousand for aligning Tau with PyTorch!
I submitted a PR https://github.com/pytorch/tau/pull/594 to fix the final lint other than those silenced ones.

**Comment by kwen2501** [comment]
@wanchaol can I land this one? Any concern?

**Comment by wanchaol** [comment]
> @wanchaol can I land this one? Any concern?

There're a bunch of ci failures, I need to resolve them before landing, do you need this soon? If so, I'll try to fix them soon and land it

**Comment by kwen2501** [comment]
I see, no hurry, please take your time. 
Just FYI -- the mypy errors are silenced, we can have more time to fix those.

File: check.sh [hunk 1]
```diff
@@ -65,7 +65,7 @@ flake8 pippy spmd test/spmd
 
 # mypy spmd test/spmd
 echo; echo "Running mypy ..."
-mypy spmd pippy test examples
+mypy spmd pippy examples
```
Comment (wanchaol) [inline check.sh:68]: @kwen2501 after restricting more mypy linting with pytorch, there emerges tons of failures in tests, imo we shouldn't require strong type annotation in tests as that will slow down the dev efficiency, let me know if this not look good to you

Comment (wanchaol) [reply check.sh:68]: This might also stay true for examples? I am changing mypy to only run under `pippy` and `spmd`

**Pull request merged**